kapitan/lint.py

```diff
@@ -148,3 +183,42 @@ def lint_unused_classes(inventory_path):
         logger.info("No usage found for the following {} classes:\n{}\n".format(len(class_paths), pformat(class_paths)))
 
     return checks_sum
+
+def lint_yamllint(inventory_path):
+    """ Run yamllint on all yaml files in inventory
+    Args:
+        inventory_path (string): path to your inventory/ folder
+    Yields:
+        checks_sum (int): the number of yaml lint issues found
+    """
+    logger.debug("Running yamllint for " + inventory_path)
+
+    if os.path.isfile('.yamllint'):
+        logger.info("Loading values from .yamllint found.")
+        conf = YamlLintConfig(file='.yamllint')
+    else:
+        logger.info(".yamllint not found. Using default values")
+        conf = YamlLintConfig(yamllint_config)
+
+    checks_sum = 0
+    for path in list_all_paths(inventory_path):
+        if os.path.isfile(path) and (path.endswith(".yml") or path.endswith(".yaml")):
+            with open(path, "r") as yaml_file:
+                file_contents = yaml_file.read()
+
+                try:
+                    problems = list(linter.run(file_contents, conf))
```

Comment:
add `filepath` to `list(linter.run(file_contents, conf, filepath=path))`
https://www.pydoc.io/pypi/yamllint-1.8.1/autoapi/linter/index.html#linter.linter.run

This way we can whitelist specific files from specific rules
https://yamllint.readthedocs.io/en/stable/configuration.html#ignoring-paths

Reply:
Done